plugin-server/src/worker/ingestion/team-manager.ts

```diff
@@ -122,6 +122,39 @@ export class TeamManager {
         }
     }
 
+    public async getTeamIdByToken(token: string): Promise<number | null> {
+        // TODO: avoid thundering herd on the DB for the same `token`
+        const cachedTeamId = this.tokenToTeamIdCache.get(token)
+
+        // tokenToTeamIdCache.get returns `undefined` if the value doesn't
+        // exist so we check for the value being `null` as that means we've
+        // explictly cached that the team does not exist
+        if (cachedTeamId === null) {
+            return null
+        } else if (cachedTeamId != null) {
+            const cachedTeam = this.teamCache.get(cachedTeamId)
+            if (cachedTeam?.id != null) {
+                return cachedTeam.id
+            }
+        }
+
+        const timeout = timeoutGuard(`Still running "fetchTeamIdByToken". Timeout warning after 30 sec!`)
+        try {
+            const teamId = await this.db.fetchTeamIdByToken(token)
+            if (!teamId) {
+                // explicitly cache a null to avoid
+                // unnecessary lookups in the future
+                this.tokenToTeamIdCache.set(token, null)
+                return null
+            }
```

Comment:
question: Couldn't an `api_token` be set before this cache is invalidated, thus returning null in subsequent calls?

Could be low risk if `api_token`s are set only once or rarely changed.

question: This kind of breaks with `team_id=0` which is used for the demo account, as 0 is falsy. Should we account for that?

Reply:
This was an almost copy and paste from the existing `fetchTeamByToken`, i.e. trying not to change from existing functionality, but I'll adjust to address the questions.

Reply:
I've removed this check now. It doesn't look like it was doing anything anyhow as the resulting code path has the same side-effect, except for the case where `team_id=0` case which was not desired anyhow.

I'm unsure about the invalidation case, but given it is not a regression I'd like to get in as is.

Reply:
Makes sense, thanks for addressing :handshake: